### PR TITLE
Formular for Joshua Decoder v6.0.4

### PR DESCRIPTION
**Pull request by lewismc**
This is a clean pull request for including a Formula for Joshua Decoder. It is updated from #44384 
Thank you

### CODE REVIEW

**Comment by chrismattmann** [comment]
@lewismc do we know the issues here?

**Comment by lewismc** [comment]
Yep... there are some bugs in Joshua which I've fixed. I think I may need to update this PR to accommodate those.

**Comment by lewismc** [comment]
Atually @chrismattmann on second thoughts I don't know.
Jenkins output looks to have failed not due to Joshua
http://bot.brew.sh/job/Homebrew%20Pull%20Requests/34025/console

**Comment by lewismc** [comment]
@bfontaine I am unsure as to what has gone wrong here? Can you possibly provide insight into how I can get tests passing here please? Thanks in advance.

File: Library/Formula/joshua-v.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  option "with-es-en-phrase-pack", "Build with Spanish–English phrase-based model [1.9 GB]."
+  option "with-ar-en-phrase-pack", "Build with Arabic–English phrase-based model [2.4 GB]." 
+
+  depends_on "ant" => :build
+  depends_on "boost" => :build
+  depends_on :python => :build if MacOS.version <= :snow_leopard
+
+  resource "es-en-phrase-pack" do
+    url "http://cs.jhu.edu/~post/language-packs/language-pack-es-en-phrase-2015-03-06.tgz"
+    sha256 "213E05BBDCFBFA05B31E263C31F10A0315695FEE26C2F37B0A78FB918BAD9B5D"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: Please use a lowercased `sha256` here and for the resource below.

**Comment by bfontaine** [comment]
@lewismc See http://bot.brew.sh/job/Homebrew%20Pull%20Requests/34025/version=yosemite/testReport/junit/brew-test-bot/yosemite/fetch_joshua_v/

```
==> Downloading http://cs.jhu.edu/~post/files/joshua-v6.0.4.tgz
Warning: Formula reports different SHA256: 0cf6674dc73f30e49a22179e90182492f8e8eea46710b505edb9acfb4d953d9a
Downloaded to: /Library/Caches/Homebrew/joshua-v-6.0.4.tgz
SHA1: d240cc2b09b01feba03f0ac5d46b4c8d81ce01b1
SHA256: 0e3b08c80303106599aa404bbba5c801cc5542779bef828d35ff80288190c273
````

The sha256 checksum seems wrong. You can check it with:
```
curl -sL 'http://cs.jhu.edu/~post/files/joshua-v6.0.4.tgz' | shasum -a 256
```

File: Library/Formula/joshua-v.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      resource("ar-en-phrase-pack").stage { share.install "language-pack-ar-en-phrase-2015-03-18", share/"joshualanguagepacks" }
+    elsif build.with? "es-en-phrase-pack"
+      resource("es-en-phrase-pack").stage { share.install "language-pack-es-en-phrase-2015-03-06", share/"joshualanguagepacks" }
+    elsif build.with? "ar-en-phrase-pack"
+      resource("ar-en-phrase-pack").stage { share.install "language-pack-ar-en-phrase-2015-03-18", share/"joshualanguagepacks" }
+    end
+  end
+
+  test do    
+    system "mkdir -p ~/models/bn-en"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: The test is executed in a temporary directory; you don’t need to create a new one.

File: Library/Formula/joshua-v.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    system "cd ~/models/bn-en"
+    system "curl -L https://github.com/joshua-decoder/indian-parallel-corpora/tarball/master > indian-languages.tgz"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: Please use a `resource` block for this.

File: Library/Formula/joshua-v.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 "213E05BBDCFBFA05B31E263C31F10A0315695FEE26C2F37B0A78FB918BAD9B5D"
+  end
+
+  resource "ar-en-phrase-pack" do
+    url "http://cs.jhu.edu/~post/language-packs/language-pack-ar-en-phrase-2015-03-18.tgz"
+    sha256 "2B6665B58B11E4C25D48191D3D5B62B7C591851A9767B14F9CCEBF1951FDDF90"
+  end
+
+  def install
+    system "ant"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: Where does this install Joshua?

File: Library/Formula/joshua-v.rb (hunk 1 above)
Comment (lewismc) [inline Library/Formula/joshua-v.rb]: done

File: Library/Formula/joshua-v.rb (hunk 2 above)
Comment (lewismc) [inline Library/Formula/joshua-v.rb]: done

File: Library/Formula/joshua-v.rb (hunk 3 above)
Comment (lewismc) [inline Library/Formula/joshua-v.rb]: done

File: Library/Formula/joshua-v.rb (hunk 4 above)
Comment (lewismc) [inline Library/Formula/joshua-v.rb]: Updated PR installs it into the Cellar

File: Library/Formula/joshua-v.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
+class JoshuaV < Formula
+  desc "Joshua is an open-source statistical machine translation decoder."
+  homepage "http://joshua-decoder.org/"
+  url "http://cs.jhu.edu/~post/files/joshua-v6.0.4.tgz"
+  sha256 "0cf6674dc73f30e49a22179e90182492f8e8eea46710b505edb9acfb4d953d9a"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: How did you get this checksum?

Comment (lewismc) [inline Library/Formula/joshua-v.rb]: As follow
http://apache.org/dev/release-signing.html#sha-checksum
```
  $ gpg --print-md SHA256 [fileName] > [fileName].sha
```

Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: I get a different checksum:

```
$ shasum -a 256 < joshua-v6.0.4.tgz
0e3b08c80303106599aa404bbba5c801cc5542779bef828d35ff80288190c273  -
$ gpg --print-md SHA256 joshua-v6.0.4.tgz
joshua-v6.0.4.tgz: 0E3B08C8 03031065 99AA404B BBA5C801 CC554277 9BEF828D
                   35FF8028 8190C273
```

Comment (lewismc) [inline Library/Formula/joshua-v.rb]: This was the original checksum I calculated and was asked to lowercase everything. Which format o you want? I can update the PR to accomodate if you prefer.

**Comment by lewismc** [comment]
@bfontaine I updated the PR based on feedback from 
```
lmcgibbn@LMC-032857 /usr/local(joshua) $ brew audit joshua-v --strict --online
==> brew style joshua-v

1 file inspected, no offenses detected
```

File: Library/Formula/joshua-v.rb (hunk 5 above)
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: We use lowercased checksums; what I mean is that the checksum you wrote is different from the one I got:

`0cf6674dc73f30e49a22179e90182492f8e8eea46710b505edb9acfb4d953d9a` v.s.
`0e3b08c80303106599aa404bbba5c801cc5542779bef828d35ff80288190c273`

This is also the one the bot gets:
```
==> Downloading http://cs.jhu.edu/~post/files/joshua-v6.0.4.tgz
Downloaded to: /Library/Caches/Homebrew/joshua-v-6.0.4.tgz
SHA1: d240cc2b09b01feba03f0ac5d46b4c8d81ce01b1
SHA256: 0e3b08c80303106599aa404bbba5c801cc5542779bef828d35ff80288190c273
```

Are you behind a proxy that might modify HTTP responses?

Comment (lewismc) [inline Library/Formula/joshua-v.rb]: @bfontaine possibly. I've updated it now and double checked all of the others. I checked all SHA256 with
```
shasum -a 256 [file name]
```
Thanks for review.

File: Library/Formula/joshua-v.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
+class JoshuaV < Formula
+  desc "Joshua is an open-source statistical machine translation decoder."
+  homepage "http://joshua-decoder.org/"
+  url "http://cs.jhu.edu/~post/files/joshua-v6.0.4.tgz"
```
Comment (bfontaine) [inline Library/Formula/joshua-v.rb]: You can use `https://` for this `url`.

**Comment by lewismc** [comment]
Closing this PR as it has been superseded by #45746

**Pull request closed without merging**